### DISCUSSION TITLE
How to switch to wasm Oqtane?

### DISCUSSION THREAD

**Discussion by ghost** [oqtane/oqtane.framework#855] [Q&A]
I changed the runtime option to Client in asppsettings.json and rebuilt the app and I don't see any dll's going to the browser. It seems it's still using server side Oqtane.

I made the client project the startup project and I got no web page.

How does one switch to the client side of Oqtane? I can't find documentation on this.

**Comment by hishamco** [comment]
No need to make the client project the startup project, just change the `Runtime` in `appsetting.json` to `WebAssembly` and run the server app

**Comment by PizzaConsole** [comment]
>  I don't see any dll's going to the browser

What dll's would you expect to go to the browser? When you swap to WebAssembly, it is Hosted. So the files in the Oqtane.Client project are the only files that are in the browser. Unless you use the `IClientStartup` on a file in the Oqtane.Server project. Basically Oqtane.Server is always server-side.

**Comment by sbwalker** [comment]
@thenrich2009 when you are running the application on WebAssembly it does download the Client DLLs from the Server to the Client. It used to be easier to see this in the Network tab of the browser tools but in recent versions of Blazor WebAssembly it hard to capture due to caching and compression. Oqtane has to take responsibility for downloading any Module or Theme DLLs as they are not statically linked to the framework - so you need to look for a call to "load" in your Network tab:

http://localhost:44357/~/api/Installation/load

If you look at the Oqtane.Server\Controllers\InstallationController.cs you will see that it gathers the specific DLLs, adds them to a ZIP file, and then sends the ZIP file to the browser ( and caches the ZIP file on the server so that future clients can reuse it ). On the client side it unpacks the ZIP file and loads the DLLs into the browser AppDomain.

**Reply by PizzaConsole** [reply]
Like I mentioned in my reply, this is assuming there are Assemblies to even load to the browser. Meaning you have to be using the interfaces `IModuleControl` or `IThemeControl` or `IClientStartup` not in the "Oqtane.Client" project for it to even have assemblies to load.

```csharp
     // get list of assemblies which should be downloaded to browser
     var assemblies = AppDomain.CurrentDomain.GetOqtaneClientAssemblies();

     public static IEnumerable<Assembly> GetOqtaneClientAssemblies(this AppDomain appDomain)
        {
            return appDomain.GetOqtaneAssemblies()
                .Where(a => a.GetTypes<IModuleControl>().Any() || a.GetTypes<IThemeControl>().Any() || a.GetTypes<IClientStartup>().Any())
                .Where(a => Utilities.GetFullTypeName(a.GetName().Name) != "Oqtane.Client");
        }
```

**Reply by ghost** [reply]
Thanks for the replies. The whole issue is because I was using 'Client' as a value for runtime in appsettings.json instead of 'WebAssembly'. The app ran fine, in server mode it seems,  instead of erroring about the wrong value so I thought it was running in WebAssembly mode. That threw me off.

**Reply by hishamco** [reply]
@thenrich2009 I already mentioned [that](https://github.com/oqtane/oqtane.framework/discussions/855#discussioncomment-117959) 6 days ago :)
